src/main.c

```diff
@@ -2408,13 +2409,16 @@ void handle_mouse_input() {
     }
 }
 
-void handle_movement(double dt) {
+void handle_movement(double dt, bool *allow_auto_camera_keyPress_ptr, bool *camera_auto_enabled_ptr)
```

Comment:
I am not sure we continue passing parameters into handle_movement. If others need to pass in pointers for similar reasons, we can easily reach a situation where we're passing in 20+ parameters. It may be better to create a struct to hold some of these values.